src/extensions/default/JavaScriptRefactoring/RenameIdentifier.js

```diff
@@ -0,0 +1,154 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var EditorManager        = brackets.getModule("editor/EditorManager"),
+        ScopeManager         = brackets.getModule("JSUtils/ScopeManager"),
+        Session              = brackets.getModule("JSUtils/Session"),
+        MessageIds           = brackets.getModule("JSUtils/MessageIds"),
+        CommandManager       = brackets.getModule("command/CommandManager"),
+        Menus                = brackets.getModule("command/Menus"),
+        Strings              = brackets.getModule("strings");
+        //Commands
+    var refactorRename          = "javascript.renamereference";
+
+    var session             = null;  // object that encapsulates the current session state
+
+    //Create new session
+    function initializeSession(editor) {
+        session = new Session(editor);
+    }
+
+    //Post message to tern node domain that will request tern server to find refs
+    function getRefs(fileInfo, offset) {
+        ScopeManager.postMessage({
+            type: "getRefs",
+            fileInfo: fileInfo,
+            offset: offset
+        });
+
+        return ScopeManager.addPendingRequest(fileInfo.name, offset, MessageIds.TERN_REFS);
+    }
+
+    //Create info required to find reference
+    function requestFindRefs(session, document, offset) {
+        var path    = document.file.fullPath,
+            fileInfo = {
+                type: MessageIds.TERN_FILE_INFO_TYPE_FULL,
+                name: path,
+                offsetLines: 0,
+                text: ScopeManager.filterText(session.getJavascriptText())
+            };
+
+        var ternPromise = getRefs(fileInfo, offset);
+
+        return {promise: ternPromise};
+    }
+
+    //Do rename of identifier which is at cursor
+    function handleRename() {
+        var editor = EditorManager.getActiveEditor(),
```

Comment:
Are you going to allow rename inside quick edit session?

Reply:
Yes. Rename is working great in quick edit as well for local variables. Renaming function calls could be a problem. According to me we should allow rename in quick edit as well. Please let me know if want me to disable rename in quick edit.

Reply:
OK. I think it is better to disable rename for quick edit. There are chances that once you invoke a quick edit on a function name, we only show the contents of the function. But there may be variables whose scope is outside the function and I am not sure how you are going to show the feedback for renaming variable references which are not shown inside quick edit.

Reply:
Let us do this. I will merge this PR. But please raise another PR addressing this concern.

Reply:
Sure @nethip I will do that. Thanks for suggestion.